domains/client/domain-operator/src/operator.rs

```diff
@@ -178,19 +189,136 @@ where
+                    if info.best_hash == info.genesis_hash {
+                        info!("Starting domain snap sync...");
+
+                        let result = snap_sync(sync_params).await;
+
+                        match result {
+                            Ok(_) => {
+                                info!("Domain snap sync completed.");
+                            }
+                            Err(err) => {
+                                error!(%err, "Domain snap sync failed.");
+
+                                // essential task failed
+                                return;
+                            }
+                        };
+                    } else {
+                        debug!("Snap sync can only work with genesis state, skipping");
+
+                        // essential task failed
+                        return;
+                    }
+
+                    // Don't exit essential task.
+                    pending().await
+                }
+            };
+
+            spawn_essential.spawn_essential("domain-sync", None, Box::pin(domain_sync_task));
+        }
+
+        let start_worker_task = {
+            let consensus_client = params.consensus_client.clone();
+            let spawn_essential = spawn_essential.clone();
+            let bundle_processor = bundle_processor.clone();
+            async move {
+                // Wait for the target block to import if we are snap syncing
+                if let Some(ref snap_sync_orchestrator) = snap_sync_orchestrator {
+                    let mut target_block_receiver =
+                        snap_sync_orchestrator.consensus_snap_sync_target_block_receiver();
+
+                    let target_block_number = match target_block_receiver.recv().await {
+                        Ok(target_block) => target_block,
+                        Err(err) => {
+                            error!(?err, "Snap sync failed: can't obtain target block.");
+                            return Err(sp_consensus::Error::Other(
+                                "Snap sync failed: can't obtain target block.".into(),
+                            ));
+                        }
+                    };
+
+                    // Wait for Subspace block importing notifications
+                    let block_importing_notification_stream =
+                        &mut params.operator_streams.block_importing_notification_stream;
+
+                    while let Some((block_number, mut acknowledgement_sender)) =
+                        block_importing_notification_stream.next().await
+                    {
+                        trace!(%block_number, "Acknowledged block import from consensus chain.");
+                        if acknowledgement_sender.send(()).await.is_err() {
+                            return Err(sp_consensus::Error::Other(
+                                format!("Can't acknowledge block import #{}", block_number).into(),
+                            ));
```

Comment:
We should print an error log here. Also, as the error of this whole future will be discarded by `start_worker_task.map(|_| ())` so makes no sense to return the error.